### PR TITLE
Merge `memory_set` into this repo

### PR DESCRIPTION
**Pull request by aarkegz**

### CODE REVIEW

**Comment by aarkegz** [comment]
Can GitHub just ff-merge this pr to keep commit history from `memory_set`?

**Comment by aarkegz** [comment]
> Can GitHub just ff-merge this pr to keep commit history from `memory_set`?

I can squash commits after the merge commit manually once it's ready to merge.

**Comment by hky1999** [comment]
I think it's ok to keep commit history from `memory_set` and use a dependent commit to represent the process of moving `memory_set` into this repository.

File: memory_set/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# memory_set
+
+[![Crates.io](https://img.shields.io/crates/v/memory_set)](https://crates.io/crates/memory_set)
+[![Docs.rs](https://docs.rs/memory_set/badge.svg)](https://docs.rs/memory_set)
+[![CI](https://github.com/arceos-org/memory_set/actions/workflows/ci.yml/badge.svg?branch=main)](https://github.com/arceos-org/memory_set/actions/workflows/ci.yml)
```
Comment (hky1999) [inline memory_set/README.md]: modify url

**Comment by hky1999** [comment]
@equation314 Is it ok to port `memory_set` into this repo first and rename this repo to names like `axmm_crates`?

File: memory_addr/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "memory_addr"
+version = "0.2.1"
```
Comment (equation314) [inline memory_addr/Cargo.toml]: Use workspace version and make both crates the same version?

File: memory_set/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+edition.workspace = true
+authors.workspace = true
+license.workspace = true
+homepage.workspace = true
+repository.workspace = true
+categories.workspace = true
+
+[dependencies]
+memory_addr = "0.2"
```
Comment (equation314) [inline memory_set/Cargo.toml:16]: `memory_addr = { workspace = true }`

**Comment by equation314** [comment]
> @equation314 Is it ok to port `memory_set` into this repo first and rename this repo to names like `axmm_crates`?

Ok, there are more commits in this repo.

File: memory_set/Cargo.toml (hunk 3 above)
Comment (hky1999) [reply memory_set/Cargo.toml:16]: @aarkegz told you (manually dog head)

File: memory_addr/Cargo.toml (hunk 2 above)
Comment (aarkegz) [reply memory_addr/Cargo.toml]: OK

File: memory_set/Cargo.toml (hunk 3 above)
Comment (aarkegz) [reply memory_set/Cargo.toml:16]: I don't think it's worth it to promote this dependency to workspace level as it's used only by `memory_set`, just keep it here...

File: memory_addr/Cargo.toml (hunk 2 above)
Comment (aarkegz) [reply memory_addr/Cargo.toml]: Updated

File: memory_set/README.md (hunk 1 above)
Comment (aarkegz) [reply memory_set/README.md]: Fixed

**Review by hky1999 — DISMISSED** [review]
LGTM

**Comment by aarkegz** [comment]
Commits squashed

**Comment by aarkegz** [comment]
Updates about `memory_set` are ready and can be pushed once this PR is merged.

**Comment by equation314** [comment]
You should also update the CI workflow file to adopt the multi-crate repo, see in the page_table_multiarch crate.

**Comment by aarkegz** [comment]
> You should also update the CI workflow file to adopt the multi-crate repo, see in the page_table_multiarch crate.

Yes, it works now fine. See [here](https://arceos-hypervisor.github.io/memory_addr/) for doc demo. `cargo fmt/clippy/test/build` also works.

**Comment by equation314** [comment]
> > You should also update the CI workflow file to adopt the multi-crate repo, see in the page_table_multiarch crate.
> 
> 
> 
> Yes, it works now fine. See [here](https://arceos-hypervisor.github.io/memory_addr/) for doc demo. `cargo fmt/clippy/test/build` also works.

The doc page in multi-crate repo is slightly different from the page jn single-crate repo. In multi-crate repo there is an index page. Please refer to https://github.com/arceos-org/page_table_multiarch/blob/main/.github/workflows/ci.yml#L43 and http://arceos.org/page_table_multiarch.

**Comment by aarkegz** [comment]
> > > You should also update the CI workflow file to adopt the multi-crate repo, see in the page_table_multiarch crate.
> > 
> > 
> > Yes, it works now fine. See [here](https://arceos-hypervisor.github.io/memory_addr/) for doc demo. `cargo fmt/clippy/test/build` also works.
> 
> The doc page in multi-crate repo is slightly different from the page jn single-crate repo. In multi-crate repo there is an index page. Please refer to https://github.com/arceos-org/page_table_multiarch/blob/main/.github/workflows/ci.yml#L43 and http://arceos.org/page_table_multiarch.

OK, I'll fix it. Can we merge this PR asap after I fix this?

**Comment by aarkegz** [comment]
> In multi-crate repo there is an index page

Fixed [here](https://arceos-hypervisor.github.io/memory_addr/).

**Review by equation314 — APPROVED** [review]

**Pull request merged**